governance/voter-weight-addin/program/src/instruction.rs

```diff
@@ -2,53 +2,48 @@
 
 use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
 use solana_program::{
+    clock::Slot,
     instruction::{AccountMeta, Instruction},
     pubkey::Pubkey,
     system_program,
 };
+use spl_governance::addins::voter_weight::VoterWeightAction;
 
 /// Instructions supported by the VoterWeightInstruction addin program
 #[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
 #[allow(clippy::large_enum_variant)]
 pub enum VoterWeightAddinInstruction {
-    /// Revises voter weight providing up to date voter weight
+    /// Sets up VoterWeightRecord owned by the program
     ///
     /// 0. `[]` Governance Program Id
     /// 1. `[]` Realm account
     /// 2. `[]` Governing Token mint
     /// 3. `[]` TokenOwnerRecord
     /// 4. `[writable]` VoterWeightRecord
-    Revise {},
-
-    /// Deposits governing token
-    /// 0. `[]` Governance Program Id
-    /// 1. `[]` Realm account
-    /// 2. `[]` Governing Token mint
-    /// 3. `[]` TokenOwnerRecord
-    /// 4. `[writable]` VoterWeightRecord
     /// 5. `[signer]` Payer
     /// 6. `[]` System
-    Deposit {
-        /// The deposit amount
+    SetupVoterWeightRecord {
```

Comment:
I'm a much bigger fan of this! Very clean.  My apologies for not remembering all of this, but previously when you did `deposit` and `withdraw`, you weren't actually moving tokens, correct?  Which is what makes this interface possible?

Reply:
My initial idea was to make the `addin` a fully working example but I've never implemented that and backtracked from that concept in this PR. Now the `addin` is just a programable mock program which I think is better. It doesn't confuse any longer  and also allows us to test spl-governance with explicit responses from the `addin`